src/github.com/getlantern/flashlight/client/pluggable_transports.go

```diff
@@ -0,0 +1,86 @@
+package client
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net"
+
+	"github.com/Yawning/obfs4/transports/obfs4"
+	"github.com/getlantern/keyman"
+	"github.com/getlantern/tlsdialer"
+
+	"git.torproject.org/pluggable-transports/goptlib.git"
+)
+
+type dialFN func() (net.Conn, error)
+type dialFactory func(*ChainedServerInfo, string) (dialFN, error)
+
+var pluggableTransports = map[string]dialFactory{
+	"":      defaultDialFactory,
+	"obfs4": obfs4DialFactory,
+}
+
+func defaultDialFactory(s *ChainedServerInfo, deviceID string) (dialFN, error) {
+	netd := &net.Dialer{Timeout: chainedDialTimeout}
+	forceProxy := ForceChainedProxyAddr != ""
+	addr := s.Addr
+	if forceProxy {
+		log.Errorf("Forcing proxying to server at %v instead of configured server at %v", ForceChainedProxyAddr, s.Addr)
+		addr = ForceChainedProxyAddr
+	}
+
+	if s.Cert == "" && !forceProxy {
+		log.Error("No Cert configured for chained server, will dial with plain tcp")
+		return func() (net.Conn, error) {
+			return netd.Dial("tcp", addr)
+		}, nil
+	}
+	log.Trace("Cert configured for chained server, will dial with tls over tcp")
+	cert, err := keyman.LoadCertificateFromPEMBytes([]byte(s.Cert))
+	if err != nil {
```

Comment:
I know this is not new, but I think we should probably make sure to log an error when there's an issue decoding the cert here. That's popped up a few times in production, and it's taken awhile to notice I think partly because it's not appearing in logs @oxtoacart.